src/transformers/models/llama/modeling_llama.py

```diff
@@ -89,11 +89,11 @@ def _expand_mask(mask: torch.Tensor, dtype: torch.dtype, tgt_len: Optional[int]
     bsz, src_len = mask.size()
     tgt_len = tgt_len if tgt_len is not None else src_len
 
-    expanded_mask = mask[:, None, None, :].expand(bsz, 1, tgt_len, src_len).to(dtype)
+    expanded_mask = mask[:, None, None, :].expand(bsz, 1, tgt_len, src_len).transpose(-2,-1).to(dtype)
 
     inverted_mask = 1.0 - expanded_mask
 
-    return inverted_mask.masked_fill(inverted_mask.to(torch.bool), torch.finfo(dtype).min)
+    return inverted_mask.masked_fill(inverted_mask.to(torch.bool), float("-inf"))
```

Comment:
I do confirm that the attention mask ends up having -inf and `torch.finfo(dtype).min` (-65500 or something like this for bfloat16) which corrupts the computation. But I don't think the solution is to cast everything to `float(-inf)`. Though it's better than having to clamp at each forward, does that not introduce overflow?

Reply:
Hi @ArthurZucker, thanks for the review. Sorry for the late reply since I was out for a vacation.

What do you mean by `overflow`? I will do cast here [1]

[1] https://github.com/huggingface/transformers/pull/26421/files#diff-06392bad3b9e97be9ade60d4ac46f73b6809388f4d507c2ba1384ab872711c51R392